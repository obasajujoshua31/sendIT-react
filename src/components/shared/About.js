import React from "react";

const About = () => {
  return (
    <React.Fragment>
      <div className="about-us">
        <div className="about-us-header">
          <div className="about-us-header-content">
            <h1>About SendIT Courier Services</h1>
            <p>
              Our mission is ensure that parcels are sent smoothly to various
              destinations across Nigeria.
            </p>
          </div>
        </div>
        <div className="about-us-body-heading">
          <h1>WHO WE ARE</h1>
        </div>
        <div className="about-us-body">
          <div className="image-container image-1">
            <img src="/src/assets/rawpixel-799527-unsplash.jpg" />
            <h3>History of SendIT</h3>
            <p>Send It began some 15 years ago at Banana Island, Lagos.</p>
          </div>
          <div className="image-container image-2">
            <img src="/src/assets/images.jpeg" />
            <h3>Motivation</h3>
            <p>
              Due to several complains regarding existing Courier Services,
              SendIT inc was formed to solve the problem
            </p>
          </div>
          <div className="image-container image-3">
            <img src="/src/assets/quickorder-301139-unsplash.jpg" />
            <h3>Vision</h3>
            <p>To be among the top leading Courier Services in Africa</p>
          </div>
                  <div className="image-container image-3">
                      <img src="/src/assets/rawpixel-783423-unsplash.jpg" />
                      <h3>Mission</h3>
                      <p>By Prompt Delivery of parcels to various destination with the satisfaction of customers in mind.</p>
                  </div>
        </div>
      </div>
    </React.Fragment>
  );
};

export default About;
