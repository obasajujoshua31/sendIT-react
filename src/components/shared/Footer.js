import React, { Fragment } from "react";

const Footer = () => {
  return (
    <Fragment>
      <footer>
        <p>SendIT &copy; 2019 Obasaju Joshua</p>
      </footer>
    </Fragment>
  );
};

export default Footer;
