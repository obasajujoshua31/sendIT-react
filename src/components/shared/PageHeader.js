import React from "react";

const PageHeader = () => {
  return (
    <div>
      <div className="container">
        <header className="service-header">
          <div className="service-brand-name">
            <h1 style={spanStyle}>
              <span>SendIT</span>SERVICES
            </h1>
          </div>
          <nav>
            <div className="service-group">
              <span className="service-span">
                <i className="fas fa-phone" />
              </span>
              <div className="service-phone-content">
                <p>078987 -87444</p>
                <p className="service-email">info@sendit.com</p>
              </div>
              <div className="service-group">
                <span className="service-span">
                  <i className="fas fa-home" />
                </span>
                <div className="service-phone-content">
                  <p>Street 43, Ikoyi Road Lagos</p>
                  <p className="service-email">23 Rail way, Ikeja.</p>
                </div>
              </div>
            </div>
          </nav>
        </header>
      </div>
    </div>
  );
};

const spanStyle = {
  color: "#846975"
};

export default PageHeader;
