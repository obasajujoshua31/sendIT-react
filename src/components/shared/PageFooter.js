import React from "react";
import { Link, Route, BrowserRouter as Router } from "react-router-dom";
import About from "../shared/About";


const PageFooter = () => {
  return (
  
      <React.Fragment>
        <div className="page-footer">
          <h1>  
            <Link to="/about">WHY CHOOSE US?</Link>
          </h1>
        </div>
      </React.Fragment>
  );
};

export default PageFooter;
