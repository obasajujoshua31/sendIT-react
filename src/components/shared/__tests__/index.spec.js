import React from "react";
import { shallow } from "enzyme";
import Footer from "../Footer";
import About from "../About";
import PageFooter from "../PageFooter";
import PageHeader from "../PageHeader";

describe("Shared Components Test", () => {
  describe("<Footer>", () => {
    const wrapper = shallow(<Footer />);
    it("should render a footer component", () => {
      expect(wrapper).toMatchSnapshot();
    });
    it("should contain a <p> tag", () => {
      expect(wrapper.find("p").length).toEqual(1);
    });
  });

  describe("<About>", () => {
    const wrapper = shallow(<About />);
    it("should render about component", () => {
      expect(wrapper).toMatchSnapshot();
    });
    it("should contain 2 <h1> tag", () => {
      expect(wrapper.find("h1").length).toEqual(2);
    });
    it("should contain 4 <h3> tag", () => {
      expect(wrapper.find("h3").length).toEqual(4);
    });
    it("should contain 5 <p> tag", () => {
      expect(wrapper.find("p").length).toEqual(5);
    });
    it("should contain 4 <img> tag", () => {
      expect(wrapper.find("img").length).toEqual(4);
    });
    it("should contain 9 <div> tag", () => {
      expect(wrapper.find("div").length).toEqual(9);
    });
  });

  describe("<PageFooter>", () => {
    const wrapper = shallow(<PageFooter />);
    it("should render a page footer component", () => {
      expect(wrapper).toMatchSnapshot();
    });
    it("should contain 1 <div> tag", () => {
      expect(wrapper.find("div").length).toEqual(1);
    });
    it("should contain 1 <h1> tag", () => {
      expect(wrapper.find("h1").length).toEqual(1);
    });
    it("should contain 1 <Link> tag", () => {
      expect(wrapper.find("Link").length).toEqual(1);
    });
  });

  describe("<PageHeader>", () => {
    const wrapper = shallow(<PageHeader />);
    it("should render a page footer component", () => {
      expect(wrapper).toMatchSnapshot();
    });
    it("should contain 7 <div> tag", () => {
      expect(wrapper.find("div").length).toEqual(7);
    });
    it("should contain 1 <h1> tag", () => {
      expect(wrapper.find("h1").length).toEqual(1);
    });
    it("should contain 3 <span> tag", () => {
      expect(wrapper.find("span").length).toEqual(3);
    });
    it("should contain 1 <header> tag", () => {
      expect(wrapper.find("header").length).toEqual(1);
    });
    it("should contain 1 <nav> tag", () => {
      expect(wrapper.find("nav").length).toEqual(1);
    });
    it("should contain 2 <i> tag", () => {
      expect(wrapper.find("i").length).toEqual(2);
    });
    it("should contain 4 <p> tag", () => {
      expect(wrapper.find("p").length).toEqual(4);
    });
  });
});
