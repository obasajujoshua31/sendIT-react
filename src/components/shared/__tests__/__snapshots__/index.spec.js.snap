// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Components Test <About> should render about component 1`] = `
<Fragment>
  <div
    className="about-us"
  >
    <div
      className="about-us-header"
    >
      <div
        className="about-us-header-content"
      >
        <h1>
          About SendIT Courier Services
        </h1>
        <p>
          Our mission is ensure that parcels are sent smoothly to various destinations across Nigeria.
        </p>
      </div>
    </div>
    <div
      className="about-us-body-heading"
    >
      <h1>
        WHO WE ARE
      </h1>
    </div>
    <div
      className="about-us-body"
    >
      <div
        className="image-container image-1"
      >
        <img
          src="/src/assets/rawpixel-799527-unsplash.jpg"
        />
        <h3>
          History of SendIT
        </h3>
        <p>
          Send It began some 15 years ago at Banana Island, Lagos.
        </p>
      </div>
      <div
        className="image-container image-2"
      >
        <img
          src="/src/assets/images.jpeg"
        />
        <h3>
          Motivation
        </h3>
        <p>
          Due to several complains regarding existing Courier Services, SendIT inc was formed to solve the problem
        </p>
      </div>
      <div
        className="image-container image-3"
      >
        <img
          src="/src/assets/quickorder-301139-unsplash.jpg"
        />
        <h3>
          Vision
        </h3>
        <p>
          To be among the top leading Courier Services in Africa
        </p>
      </div>
      <div
        className="image-container image-3"
      >
        <img
          src="/src/assets/rawpixel-783423-unsplash.jpg"
        />
        <h3>
          Mission
        </h3>
        <p>
          By Prompt Delivery of parcels to various destination with the satisfaction of customers in mind.
        </p>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Shared Components Test <Footer> should render a footer component 1`] = `
<Fragment>
  <footer>
    <p>
      SendIT Â© 2019 Obasaju Joshua
    </p>
  </footer>
</Fragment>
`;

exports[`Shared Components Test <PageFooter> should render a page footer component 1`] = `
<Fragment>
  <div
    className="page-footer"
  >
    <h1>
      <Link
        replace={false}
        to="/about"
      >
        WHY CHOOSE US?
      </Link>
    </h1>
  </div>
</Fragment>
`;

exports[`Shared Components Test <PageHeader> should render a page footer component 1`] = `
<div>
  <div
    className="container"
  >
    <header
      className="service-header"
    >
      <div
        className="service-brand-name"
      >
        <h1
          style={
            Object {
              "color": "#846975",
            }
          }
        >
          <span>
            SendIT
          </span>
          SERVICES
        </h1>
      </div>
      <nav>
        <div
          className="service-group"
        >
          <span
            className="service-span"
          >
            <i
              className="fas fa-phone"
            />
          </span>
          <div
            className="service-phone-content"
          >
            <p>
              078987 -87444
            </p>
            <p
              className="service-email"
            >
              info@sendit.com
            </p>
          </div>
          <div
            className="service-group"
          >
            <span
              className="service-span"
            >
              <i
                className="fas fa-home"
              />
            </span>
            <div
              className="service-phone-content"
            >
              <p>
                Street 43, Ikoyi Road Lagos
              </p>
              <p
                className="service-email"
              >
                23 Rail way, Ikeja.
              </p>
            </div>
          </div>
        </div>
      </nav>
    </header>
  </div>
</div>
`;
